packages/http-client-csharp/emitter/src/emitter.ts

```diff
@@ -180,6 +197,61 @@ export async function $onEmit(context: EmitContext<NetEmitterOptions>) {
   }
 }
 
+/** check the dotnet sdk installation.
+ * Report diagnostic if dotnet sdk is not installed or its version does not meet prerequisite
+ * @param program The typespec compiler program
+ * @param minVersionRequisite The minimum required version
+ */
+async function validateDotnet(program: Program, minVersionRequisite: string): Promise<boolean> {
+  const parsedVersions = semver.parse(minVersionRequisite);
+  if (!parsedVersions) {
+    Logger.getInstance().error("invalid parameter: minVersionRequisite.");
+    return false;
+  }
+  try {
+    const result = await execAsync("dotnet", ["--list-sdks"]);
+    const dotnetVersions = findDonetVersion(result.stdout) ?? findDonetVersion(result.stderr);
+    if (dotnetVersions) {
+      for (const version of dotnetVersions) {
+        if (semver.gt(version, minVersionRequisite)) return true;
```

Comment:
Not sure that it is worth using the semver dependency for this simple check. We can even make the check simpler if we don't accept an arbitrary minVersion parameter and just check that the major version is 8.

Reply:
Now we only check the major version, no need to use semver. remove the semver dependency.